be/src/exprs/agg/percentile_approx.h

```diff
@@ -79,9 +105,11 @@ class PercentileApproxAggregateFunction final
         double quantile;
         memcpy(&quantile, src.data, sizeof(double));
 
-        PercentileApproxState src_percentile;
+        PercentileApproxState src_percentile(this->get_compression_factor(ctx));
         src_percentile.targetQuantile = quantile;
         src_percentile.percentile->deserialize((char*)src.data + sizeof(double));
+
+        init_state_if_necessary(ctx, state);
         int64_t prev_memory = data(state).percentile->mem_usage();
         data(state).percentile->merge(src_percentile.percentile.get());
         data(state).targetQuantile = quantile;
```

Comment:
I believe this issue won't occur because the init_state_if_necessary function will ensure that the compression used by the current percentile and the compression of the src state are the same.